### ISSUE TITLE
About the result of the game after 250 hands.

### ISSUE THREAD

**Issue by dongcijie** [gcp/leela-zero#2058]
After 250 hands, it is not the opponent of V1.
![360 20181130110056250](https://user-images.githubusercontent.com/32858319/49266400-0ec49680-f491-11e8-8bc9-d16b664e7eed.jpg)

**Comment by zhanzhenzhen** [comment]
This is just because ELF has a sharper win/loss evaluation than LZ does. See:
https://github.com/gcp/leela-zero/issues/1791#issuecomment-417678686

**Comment by Splee99** [comment]
This means we don't have enough quality long self-play games with more than 250 moves. Those non-resign games don't help. Could we sort the self play games somehow to include more good long self-play games in training?

**Comment by l1t1** [comment]
how to draw a graph like this for other match?

**Comment by l1t1** [comment]
please attach the excel file

**Comment by roy7** [comment]
> This means we don't have enough quality long self-play games with more than 250 moves. Those non-resign games don't help. Could we sort the self play games somehow to include more good long self-play games in training?

@Splee99 I'm not sure what you mean here. If a self-play game resigns before 250 moves then Leela is pretty sure the game is already decided. I don't think anyone has tested the false resign rate lately but it shouldn't be too high hopefully based on our past tests. There's no point making games run longer if Leela is already sure one side has won.

**Comment by Splee99** [comment]
@roy7 With the noise enabled from the start lz is More likely to enter unbalanced state earlier, no matter resign is enabled or not. That means game result is determined also earlier, probably before the 200th move. Now it seems we need to train lz for more balanced situations after move 200. It could be done by selecting  more long resign games or enable noise in later stage of self play.

**Comment by alreadydone** [comment]
@trnaeo 40b requires twice many computation to generate the same amount of training data.
@Splee99 If there is no noise then moves with low policy will never be explored, receive 0 visits, and never selected even with randomness. Even worse, because they have 0 visits their policy will be further trained towards 0.

**Comment by Splee99** [comment]
@alreadydone I'm not sure you got my point. I would like to feed board positions aka game states which are beyond move 200 into the training process. These are of course generated with noise enabled. However these positions need to be reasonably balanced and not like 1% winrate to one side.

**Comment by thynson** [comment]
I've been argued before that since visit count decreased from 3200 to 1600, the current noise factor may be too high. Not just because peak bad noise lead the game to unbalanced position too early, also it make the value head became dull as from the statistics the value head believe both side will make mistake in future.

**Comment by Splee99** [comment]
Anyways, it looks like we have to modify autogtp to achieve this. In the first half the noise is disabled to allow a balanced mid game. Then the noise is enabled, maybe after the 200th move, to generate proper training data. After the game is finished, only upload the second half of the game to the sever for training.

**Comment by Ishinoshita** [comment]
For a given net, with current self-play hyper-parameters and resign enabled at 5%, game length distribution is what it is.

Increasing the proportion of high move count positions by playing no resign games has its own drawback, as, past the resignation threshold, extreme values + noise cause the search to produce poor policy training target.

IMHO, generating more good quality high move count positions would probably take to remove randomness past some value (5%?, 2%?). Games would be shorter, so we could increase the proportion of no resign games, to increase the proportion of longer games.

**Comment by alreadydone** [comment]
All matches shown on zero.sjeng.org were played under visit parity, and 40b nets should eventually beat 20b nets under time parity as well (at least at high visit count, since 40b should scale better). In fact I heard that # 193 beat ELFv1 several times in a row as black under time parity (90 seconds, 500k-1m playouts per move).
